tests/tests/operation_tests.cpp

```diff
@@ -1776,77 +1763,62 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
+            op.fee = ASSET_GOLOS(100);
+            CHECK_OP_VALID(op);
+            CHECK_PARAM_VALID(op, url, "-");
+            CHECK_PARAM_VALID(op, url, string(STEEMIT_MAX_WITNESS_URL_LENGTH, ' '));
+            CHECK_PARAM_VALID(op, url, u8"тест");
+            CHECK_PARAM_VALID(op, fee, ASSET_GOLOS(0));
+
+            BOOST_TEST_MESSAGE("--- failure when owner is empty");
+            CHECK_PARAM_INVALID(op, owner, "");
+
+            BOOST_TEST_MESSAGE("--- failure when url is empty or too long or have invalid utf8");
+            CHECK_PARAM_INVALID(op, url, "");
+            // CHECK_PARAM_INVALID(op, url, string(1+STEEMIT_MAX_WITNESS_URL_LENGTH, ' ')); // needs HF
+            CHECK_PARAM_INVALID(op, url, "\xc3\x28");
+
+            BOOST_TEST_MESSAGE("--- failure when fee in not GOLOS or negative");
+            CHECK_PARAM_INVALID(op, fee, ASSET_GBG(1));
+            CHECK_PARAM_INVALID(op, fee, ASSET_GESTS(1));
+            CHECK_PARAM_INVALID(op, fee, ASSET_GOLOS(-1));
+
+            // TODO: chain_properties_17
             validate_database();
         }
         FC_LOG_AND_RETHROW()
     }
 
-    BOOST_AUTO_TEST_CASE(witness_update_authorities) {
-        try {
-            BOOST_TEST_MESSAGE("Testing: witness_update_authorities");
-
-            ACTORS((alice)(bob));
-            fund("alice", 10000);
-
-            private_key_type signing_key = generate_private_key("new_key");
-
-            witness_update_operation op;
-            op.owner = "alice";
-            op.url = "foo.bar";
-            op.fee = ASSET("1.000 GOLOS");
-            op.block_signing_key = signing_key.get_public_key();
-
-            signed_transaction tx;
-            tx.set_expiration(db->head_block_time() + STEEMIT_MAX_TIME_UNTIL_EXPIRATION);
-            tx.operations.push_back(op);
-
-            BOOST_TEST_MESSAGE("--- Test failure when no signatures");
-            STEEMIT_REQUIRE_THROW(db->push_transaction(tx, 0), tx_missing_active_auth);
-
-            BOOST_TEST_MESSAGE("--- Test failure when signed by a signature not in the account's authority");
-            tx.sign(alice_post_key, db->get_chain_id());
-            STEEMIT_REQUIRE_THROW(db->push_transaction(tx, 0), tx_missing_active_auth);
+    BOOST_AUTO_TEST_CASE(witness_update_authorities) { try {
+        BOOST_TEST_MESSAGE("Testing: witness_update_authorities");
+        witness_update_operation op;
+        op.owner = "bob";
+        op.url = "http://localhost";
+        // op.fee = ASSET_GOLOS(1);
+        // op.block_signing_key = signing_key.get_public_key();
 
-            BOOST_TEST_MESSAGE("--- Test failure when duplicate signatures");
-            tx.signatures.clear();
-            tx.sign(alice_private_key, db->get_chain_id());
-            tx.sign(alice_private_key, db->get_chain_id());
-            STEEMIT_REQUIRE_THROW(db->push_transaction(tx, 0), tx_duplicate_sig);
+        CHECK_OP_AUTHS(op, account_name_set(), account_name_set({"bob"}), account_name_set());
 
```

Comment:
What about another checks: tx_missing_active_auth, tx_duplicate_sig, tx_irrelevant_sig? I understand that they do not refer to the functional under test. But are we ready to remove these checks in the current task (@afalaleev)?

Reply:
There are no many tests which check transaction errors. 
And I don't think that it is not a good idea to remove these checks.
@zxcat What reason was to remove them?

Reply:
That checks duplicated in some older tests and don't not actually check operation authorities, but `push_transaction` functionality. That processes is only depends on `get_required_authorities` result. And it's even not on operation level, but transaction.
IDK, is there separate `push_transaction` tests, but `operation_tests.cpp` still contains lot of places, where `tx_missing_xxx_auth`, `tx_duplicate_sig` and `tx_irrelevant_sig` checked.
It's better to move this in one place and not put to operation tests.